Shared/Tracking/Trackers/shikimori/ShikimoriApi.swift

```diff
@@ -0,0 +1,253 @@
+//
+//  ShikimoriApi.swift
+//  Aidoku
+//
+//  Created by Vova Lapskiy on 02.11.2024.
+//
+
+import Foundation
+
+class ShikimoriApi {
+    private let decoder = JSONDecoder()
+    private let encoder = JSONEncoder()
+    private let useragent = "Aidoku"
+    private let dateformat = "yyyy-MM-dd'T'HH:mm:ss.SSSxxx"
+
+    // Registered under vlapsk1y's Shikimori account
+    var oauth = OAuthClient(
+        id: "shikimori",
+        clientId: "O9LOQF35DlUZhnPmzO2WNOGZK_caeaJ0joXUTQAo9J4",
+        clientSecret: "C6-Mz-u-yx_Bka3PLCz2NJKdDsKcSGa6KqoXaMfic-8",
+        baseUrl: "https://shikimori.one/"
+    )
+}
+
+extension ShikimoriApi {
+    // MARK: custom OAuth
+
+    func getAuthenticationUrl() -> String? {
+        guard let url = URL(string: oauth.baseUrl + "oauth/authorize") else { return nil }
+        var components = URLComponents(url: url, resolvingAgainstBaseURL: true)
+        let queryItems = [
+            URLQueryItem(name: "client_id", value: oauth.clientId),
+            URLQueryItem(name: "redirect_uri", value: "aidoku://shikimori-auth"),
+            URLQueryItem(name: "response_type", value: "code"),
+            URLQueryItem(name: "scope", value: "user_rates")
+        ]
+        components?.queryItems = queryItems
+        return components?.url?.absoluteString
+    }
+
+    func getAccessToken(authCode: String) async -> OAuthResponse? {
+        guard let url = URL(string: oauth.baseUrl + "oauth/token") else { return nil }
+        let boundary = "--" + String(UUID().hashValue)
+        var request = URLRequest(url: url)
+
+        request.httpMethod = "POST"
+        request.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        request.setValue(useragent, forHTTPHeaderField: "User-Agent")
+        request.httpBody = multipart(params: [
+            "grant_type": "authorization_code",
+            "client_id": oauth.clientId,
+            "client_secret": oauth.clientSecret!,
+            "redirect_uri": "aidoku://shikimori-auth",
+            "scope": "users_rate",
+            "code": authCode
+        ], boundary: boundary)
+        oauth.tokens = try? await URLSession.shared.object(from: request)
+        oauth.saveTokens()
+        return oauth.tokens
+    }
+
+    // MARK: API Methods - Data
+
+    func search(query: String) async -> GraphQLResponse<ShikimoriMangas>? {
+        await requestGraphQL(
+            GraphQLVariableQuery(
+            query: ShikimoriQueries.searchQuery,
+            variables: ShikimoriSearchVars(search: query))
+        )
+    }
+
+    func register(_ trackId: String, _ hasReadChapters: Bool) async -> String? {
```

Comment:
i think it would be better if you used the actual parameter names, since that's the general swift convention (also applies to `update`)
```suggestion
    func register(trackId: String, hasReadChapters: Bool) async -> String? {
```